.github/workflows/linux-eic-shell.yml

```diff
@@ -69,12 +69,22 @@ jobs:
       matrix:
         CXX: [g++, clang++]
         CMAKE_BUILD_TYPE: [Release, Debug]
+        # FIXME: gcc-12 does not allow ignoring unknown-pragmas with a pragma
+        # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53431
+        CXXFLAGS: [-Werror -Wall -Wextra -Wno-unknown-pragmas -Wno-error=maybe-uninitialized]
```

Comment:
This isn't right, though. We don't want to disable maybe-uninitialized everywhere just because there is an issue in a gcc header.

Reply:
I mean, this is exactly where we will find sources of undefined behavior.

Reply:
Sorry, I was testing privately, but pushed to wrong repo.

Yes, I'm thinking to disable this warning. If compilers are buggy, I don't think we should keep the pragmas around.

Reply:
We would not need to disable the same sets of warnings on all compilers, I assume.

Reply:
The issue in functional for gcc-12 was fixed in gcc-13. It was treated as a legitimate bug. I don't think we have to expect this to happen, any more than having other compiler bugs hitting us.

The warning about false positives when combining sanitizers and initialization warnings is for user code, not targeted at gcc c++ library implementation code.

I think splitting the pipelines so only gcc or only clang runs with certain warnings could make it hard to debug for users.